src/Groups/GAPGroups.jl

```diff
@@ -216,27 +216,126 @@ function rand_pseudo(G::GAPGroup; radius::Int = 10)
   return group_element(G, GAP.Globals.PseudoRandom(GapObj(G); radius = radius)::GapObj)
 end
 
+
+# We allow arithmetic operations between two group elements with
+# *nonidentical parents*.
+# In this case, the parent of the resulting element is set according to
+# the following rules, depending on the types of the parents.
+#
+# - `PermGroup`, `PermGroup`:
+#   The operation is allowed whenever the parents have the same degree,
+#   and the parent is the symmetric group of that degree.
+#
+# - `PcGroup`, `PcGroup` and
+#   `FPGroup`, `FPGroup`:
+#   The operation is allowed whenever the parents belong to the same `GapObj`
+#   (in the sense of `==´), and the first of the two groups is taken.
+#
+# - `SubPcGroup`, `SubPcGroup` and
+#   `SubFPGroup`, `SubFPGroup`:
+#   The operation is allowed whenever the parents have the same `full_group`
+#   (in the sense of `==´), and the first of the two groups is taken.
+#
+# - `SubPcGroup`, `PcGroup` and
+#   `PcGroup`, `SubPcGroup` and
+#   `SubFPGroup`, `FPGroup` and
+#   `FPGroup`, `SubFPGroup`:
+#   The operation is allowed whenever the `full_group` of the `SubPcGroup`
+#   (`SubFPGroup`) is equal to the `PcGroup` (`FPGroup`)
+#   (in the sense of `==´), and the `PcGroup` (`FPGroup`) is taken.
+#
+# - `MatrixGroup`, `MatrixGroup`:
+#   The operation is allowed whenever the two groups have the same `degree`
+#   and `base_ring`, and the parent is the general linear group of that
+#   degree over that ring.
+#
+# - `AutomorphismGroup`, `AutomorphismGroup`:
+#   The operation is allowed whenever the two groups have the same `.G` field,
+#   and the parent is the full automorphism group of that group.
+#
+# - `DirectProductGroup`, `DirectProductGroup`:
+#   The operation is allowed whenever the two groups have the same `.L`
+#   field, and the parent is the direct product of these groups.
+#
+# For other types of groups, we throw an exception if the parent groups are
+# not equal and their `GapObj`s are not equal.
+#
+# Note that we cannot guarantee that the groups objects in question are
+# *identical*, only that their `GapObj`s are *equal*:
+# On the GAP side, the same group can be created several times,
+# and on the Oscar side, the same GAP group can be wrapped several times.
+#
+# Note also that we do not want to perform checks whether one group is perhaps
+# a subset of the other (and then one could take the bigger group as
+# the parent of the result).
+#
+function _common_parent_group(x::PermGroup, y::PermGroup)
+  x === y && return x
+  @req degree(x) == degree(y) "the groups have different degrees"
+  return symmetric_group(degree(x))
+end
+
+function _common_parent_group(x::PcGroup, y::PcGroup)
+  (x === y || GapObj(x) == GapObj(y)) && return x
```

Comment:
Does `==` for `PcGroup` do anything besides compare identity?

If so, wouldn't this be sufficient?
```suggestion
  (GapObj(x) === GapObj(y)) && return x
```

Reply:
The question is what we want to support.
Currently the following works.
```
julia> g = small_group(24, 12);

julia> gg = GAP.Globals.Group(GAP.Globals.GeneratorsOfGroup(GapObj(g)));

julia> newg = PcGroup(gg);

julia> newg == g
true
```
If we want to achieve that two `PcGroup`s are equal only if their `GapObj`s are identical then we have to change `PcGroup`. Likewise for `FPGroup`s. I do not expect difficulties from such a change.